MiniApp/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift

```diff
@@ -340,8 +335,20 @@ internal class MiniAppScriptMessageHandler: NSObject, WKScriptMessageHandler {
         }
     }
 
-    func fetchTokenDetails(callbackId: String) {
-        hostAppMessageDelegate?.getAccessToken(miniAppId: self.miniAppId) { (result) in
+    func fetchTokenDetails(callbackId: String, for requestParam: RequestParameters?) {
+        var accessTokenPermission: MASDKAccessTokenPermission?
+        if let audience = requestParam?.audience { // if we request a token for a specific audience
+            accessTokenPermission = MASDKAccessTokenPermission(audience: audience, scopes: [])
+            if let scopes = requestParam?.scopes { // we need a specific set of scopes
+                accessTokenPermission?.scopes = scopes
+            } else { return sendScopeError(callbackId: callbackId, type: .scopeError) }
+            if let accessTokenPermissions = miniAppManifest?.accessTokenPermissions { // we check that the Mini App manages scopes
+                if !(accessTokenPermission!.isPartOf(accessTokenPermissions)) { // and that these scopes are included in the Manifest
+                    return sendScopeError(callbackId: callbackId, type: .audienceError)
+                }
+            } else { return sendScopeError(callbackId: callbackId, type: .audienceError) }
```

Comment:
You are throwing the same error, is it possible to make them as one?

Reply:
yeah, maybe if I think a little more about it